tests/tests_pytorch/models/test_tpu.py

```diff
@@ -310,30 +303,37 @@ def test_tpu_cores_with_argparse(cli_args, expected):
 
 
 @RunIf(tpu=True)
-@pl_multi_process_test
 def test_tpu_reduce():
     """Test tpu spawn reduce operation."""
 
     def test_reduce(rank):
         trainer = Trainer(accelerator="tpu", devices=8)
-        # faster this way
-        reduce_ops = ["mean", "AVG", "undefined", "sum", ReduceOp.SUM, ReduceOp.MAX]
-        for reduce_op in reduce_ops:
-            if reduce_op == "undefined" or reduce_op == ReduceOp.MAX:
-                with pytest.raises(MisconfigurationException, match="TPUSpawn Strategy only support"):
-                    result = trainer.strategy.reduce(1, reduce_op)
-            else:
-                result = trainer.strategy.reduce(1, reduce_op)
+        trainer.strategy._launched = True
+
+        with pytest.raises(ValueError, match="TPUSpawnStrategy only supports"):
+            trainer.strategy.reduce(1, reduce_op="undefined")
+
+        with pytest.raises(ValueError, match="TPUSpawnStrategy only supports"):
+            trainer.strategy.reduce(1, reduce_op=ReduceOp.MAX)
+
+        # it is faster to loop over here than to parameterize the test
+        for reduce_op in ("mean", "AVG", "sum", ReduceOp.SUM):
+            result = trainer.strategy.reduce(1, reduce_op=reduce_op)
             if isinstance(reduce_op, str) and reduce_op.lower() in ("mean", "avg"):
                 assert result.item() == 1
             else:
                 assert result.item() == 8
 
+        # https://github.com/pytorch/xla/issues/1801#issuecomment-602799542
+        xm.rendezvous("end-process")
+        # https://github.com/pytorch/xla/issues/2190#issuecomment-641665358
+        if rank == 0:
+            time.sleep(1)
```

Comment:
Since this is used in multiple tests in this file, create a function with this logic?